src/test/kotlin/com/example/realworldkotlinspringbootjdbc/infra/CommentRepositoryImplTest.kt

```diff
@@ -0,0 +1,164 @@
+                    updatedAt = date,
+                    authorId = UserId(3),
+                ),
+            )
+            when (actual) {
+                is Left -> assert(false)
+                is Right -> {
+                    // List<Comment> 1 つめの比較（createdAt, updateAtはメタデータなので比較しない）
+                    assertThat(actual.value[0].id).isEqualTo(expected[0].id)
+                    assertThat(actual.value[0].body).isEqualTo(expected[0].body)
+                    assertThat(actual.value[0].authorId).isEqualTo(expected[0].authorId)
+                    // List<Comment> 2 つめの比較（createdAt, updateAtはメタデータなので比較しない）
+                    assertThat(actual.value[1].id).isEqualTo(expected[1].id)
+                    assertThat(actual.value[1].body).isEqualTo(expected[1].body)
+                    assertThat(actual.value[1].authorId).isEqualTo(expected[1].authorId)
+                    // List<Comment> 3 つめの比較（createdAt, updateAtはメタデータなので比較しない）
+                    assertThat(actual.value[2].id).isEqualTo(expected[2].id)
+                    assertThat(actual.value[2].body).isEqualTo(expected[2].body)
+                    assertThat(actual.value[2].authorId).isEqualTo(expected[2].authorId)
+                }
+            }
+        }
+
+        @Test
+        @DataSet(
+            value = [
+                "datasets/yml/given/articles.yml",
+            ],
+        )
+        fun `正常系-articles テーブルに slug に該当するが Comment 存在しなかった場合、空の Comment の List が戻り値`() {
+            /**
+             * given:
+             */
+            val commentRepository = CommentRepositoryImpl(namedParameterJdbcTemplate)
+
+            /**
+             * when:
+             */
+            val actual = commentRepository.list(Slug.newWithoutValidation("functional-programming-kotlin"))
+
+            /**
+             * then:
+             */
+            val expected = listOf<Comment>()
+            when (actual) {
+                is Left -> assert(false)
+                is Right -> assertThat(actual.value).isEqualTo(expected)
+            }
+        }
+
+        @Test
+        @DataSet(
+            value = [
+                "datasets/yml/given/empty-articles.yml",
+            ],
+        )
+        fun `準正常系-articles テーブルに slug に該当する記事がなかった場合、NotFoundError が戻り値`() {
+            /**
+             * given:
+             */
+            val commentRepository = CommentRepositoryImpl(namedParameterJdbcTemplate)
+
+            /**
+             * when:
+             */
+            val actual = commentRepository.list(Slug.newWithoutValidation("not-found-article-slug"))
+
+            /**
+             * then:
+             */
+            val expected =
+                CommentRepository.ListError.NotFoundArticleBySlug(Slug.newWithoutValidation("not-found-article-slug"))
+            when (actual) {
+                is Left -> assertThat(actual.value).isEqualTo(expected)
+                is Right -> assert(false)
+            }
```

Comment:
[imo]

以下のような感じはどうでしょう？（leftはどっちでもいいかと）

（ただ、他のテストと形を揃えることを優先しているのであれば、whenのままでもいいかと思っています）

```suggestion
            val expected =
                CommentRepository.ListError.NotFoundArticleBySlug(Slug.newWithoutValidation("not-found-article-slug")).left()
            assertThat(actual).isEqualTo(expected)
```

```suggestion
            val expected =
                CommentRepository.ListError.NotFoundArticleBySlug(Slug.newWithoutValidation("not-found-article-slug"))
            assertThat(actual).isEqualTo(expected.left())
```

Reply:
https://github.com/sunakan/realworld-kotlin-springboot-jdbc/pull/107/commits/74a56d0f201054cbdc2bd1c30ca74768854e5ed6

Reply:
https://github.com/sunakan/realworld-kotlin-springboot-jdbc/pull/107/commits/d6708b032e751f238c5f93994bc718af0a4c03b9